notebook/base/handlers.py

```diff
@@ -629,6 +629,10 @@ def set_headers(self):
         # disable browser caching, rely on 304 replies for savings
         if "v" not in self.request.arguments:
             self.add_header("Cache-Control", "no-cache")
+
+        # In case we're serving HTML/SVG, confine any Javascript to a unique
+        # origin so it can't interact with the notebook server.
+        self.set_header('Content-Security-Policy', 'sandbox allow-scripts')
```

Comment:
This also overrides any currently set `Content-Security-Policy` -- should we merge with the operator's settings?

Note: I refer to operator to mean whoever administers this deployment, whether a local server user or a jupyterhub administrator.

Reply:
Yes, good point. Do you think it's sufficient to use `add_header()` to do this, or should we get into checking any existing `Content-Security-Policy` headers?

Reply:
Can multiple CSP headers be set and do what we want? If not, we might want to override the inherited [content_security_policy property](https://github.com/jupyter/notebook/blob/master/notebook/base/handlers.py#L66) instead.

Reply:
[From MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy):

> You can use the Content-Security-Policy header more than once... Adding additional policies can only further restrict the capabilities of the protected resource

I think that means this should work, but maybe the c_s_p property is a neater way to achieve the same thing?

Reply:
Interesting, I didn't realize we could add multiple headers. How does `set_header` work when there are multiple headers with the same key from `add_header`?

Reply:
I think `set_header` blows away anything that was set before.

Reply:
My question comes down to "what's the set of headers after running the following"

```python
self.add_header("X", "y")
self.add_header("X", "z")
self.set_header("X", "x")
```

Reply:
AIUI, only `X: x` would remain.

Reply:
I'm thinking using only one header is good. For now I think the value you've set here is a very sane default and further enhancements on merging content security policies can be separate work.